linters/amp_remove_empty_lines.py

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env python
+"""
+Remove empty lines within a function.
+
+Import as:
+
+import linters.amp_remove_empty_lines as lareemli
+"""
+
+import argparse
+from typing import List
+
+import helpers.hdbg as hdbg
+import helpers.hio as hio
+import helpers.hparser as hparser
+import helpers.hstring as hstring
+import linters.action as liaction
+import linters.utils as liutils
+
+
+def update_function_blocks(file_content: str) -> List[str]:
+    """
+    Process file to find and update functions.
+
+    :param file_content: file to process
+    :return: formatted file without empty lines in functions
+    """
+    lines = file_content.splitlines()
+    if file_content.endswith("\n"):
+        lines.append("")
+    docstring_indices = set(hstring.get_docstring_line_indices(lines))
+    function_indices = set(hstring.get_function_indices(lines))
+    cleaned_file = []
+    for i, line in enumerate(lines):
+        stripped = line.strip()
+        # Skip empty lines, except in docstrings.
+        if (
```

Comment:
So complicated. Find the docstrings, find the function bodies, ... why can't it be done in a single pass? A normal function has a regular structure

```
def foobar(...)
    """
    """
    blah blah

something else
```